File: Java/Algorithms/Insertion Sort/insertion_sort.java
public class insertion_sort {
    public static void print_array(int arr[], int size) {
        for(int i=0; i<size; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    public static void insertionSort(int arr[], int size) {
        for(int i=1; i<size; i++) {
            int value = arr[i];
            int hole = i;
            while(hole>0 && arr[hole-1] > value) {
                arr[hole] = arr[hole-1];
                hole--;
            }
            arr[hole] = value;
        }
    }
    public static void main(String[] args) {
        int[] arr = new int[] { 121, 43, 11, 34, 5, 34, 65, 567, 23, 238 };
        int size = arr.length;
        System.out.println("initial array :");
        print_array(arr, size);
        System.out.println("sorted array :");
        insertionSort(arr, size);
        print_array(arr, size);
    }
}

File: C/Algorithms/Insertion Sort/insertion_sort.c
#include<stdio.h>

void printArray(int arr[], int size) {
    int i;
    for(i=0; i<size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}
void insertionSort(int arr[], int size) {
    int i;
    for(i=1; i<size; i++) {
        int current = arr[i];
        int j = i-1;
        while(j>=0 && arr[j] > current) {
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = current;
    }
}

int main() {
    int arr[10] = { 234, 34, 348, 934, 4, 12, 49, 34798, 23, 2 };
    int size = sizeof(arr)/sizeof(arr[0]);
    printf("initial array :\n");
    printArray(arr, size);
    insertionSort(arr, size);
    printf("sorted array :\n");
    printArray(arr, size);

    return 0;
}
